src/lib.rs

```diff
@@ -115,7 +123,86 @@ impl Procyon {
 
         new
     }
-
+    /// save the image
+    pub async fn save(&self, image_path: &Path) -> Result<(), Box<dyn std::error::Error>> {
+        // Spawn in background a webdriver
+        // Currently geckodriver is hardcoded but should support
+        // all webdriver compliant driver.
+        // TODO:updated to all webdriver
+        let mut webdriver_process = Command::new("geckodriver")
+            .args(&["--port", "4444"])
+            .spawn()
+            .expect("webdriver failed to start");
+        // This should also be TODO:updated to all webdriver
+        let mut caps = serde_json::map::Map::new();
+        let opts = serde_json::json!({ "args": ["--headless"]});
+        caps.insert("moz:firefoxOptions".to_string(), opts);
+
+        let mut connection_ok = false;
+        while !connection_ok {
+            thread::sleep(time::Duration::from_secs(5));
```

Comment:
This, unfortunately, means that there is no way to wait less than 5 seconds to save a plot.
Maybe start with a much smaller delay and then exponentially backoff.

Maybe it is worth splitting the method of creating a headless Browser, for instance Chromium or Firefox, from creating and storing plots in the instance.

By doing this, users can decide to call a method that first starts the Browser and then saves the plot and tears down the Browser, or to keep the instance alive and reuse it.

Of course, this looks like a future task.

Reply:
I will split it now because it will add readability and we can talk more specifically about this function signature and possible use.